zebra-state/src/service.rs

```diff
@@ -270,6 +270,95 @@ impl StateService {
             state: IterState::NonFinalized(hash),
         }
     }
+
+    /// Return an iterator over the finalized blocks of the chain
+    /// starting (and descending) from the block identified by `height`.
+    ///
+    /// The block identified by `height` is included in the chain of blocks yielded
+    /// by the iterator.
+    pub fn chain_finalized(&self, height: block::Height) -> Iter<'_> {
+        Iter {
+            service: self,
+            state: IterState::Finalized(height),
+        }
+    }
+
+    /// Return a list of blocks ahead given a list of blocks.
+    ///
+    /// 1. Make sure we have the list of blocks provided in our chain.
+    /// 2. Get a max of 500 hashes ahead of the provided tip.
+    pub fn find_chain_hashes(
+        &self,
+        block_locator: Vec<block::Hash>,
+        stop: Option<block::Hash>,
+    ) -> Option<Vec<block::Hash>> {
+        let locator_tip_hash = block_locator.first()?;
+        let locator_tip_height = self.height_by_hash(*locator_tip_hash)?;
+        let chain = self.chain_finalized(locator_tip_height);
+
+        tracing::info!(
```

Comment:
Please downgrade to trace or debug before merging
```suggestion
        tracing::trace!(
```

Reply:
Just a reminder that the log levels in this PR need to be fixed before merging.